RELEASING.md

```diff
@@ -0,0 +1,49 @@
+# Release Process
+
+## Versioning
+The versioning follows the process described at [semver.org](https://semver.org/).
+
+### Public API
+We consider the public API of this project to be whatever is included in the
+`LIBRARY_PUBLIC_HEADERS` defined in `src/CMakeLists.txt`. Currently, this includes
+all the header files in `src/mococrw/`. Changes to the files will be versioned according
+to semantic versioning scheme.
+
+## PRs & Releasing
+
+### PR Checklist
+(probably to be extended)
+
+What should be done with every PR (if no new release is created):
+* CHANGELOG is updated with the details of the change (Unreleased section)
+
+### Releasing Checklist
+(probably to be extended)
+
+Check if the following steps have been performed **before** creating a new release. This includes
+all kinds of releases (Major, Minor and Patch releases)
+* README is updated if necessary
+* CHANGELOG is updated (add the new release)
+* Library .so version is increased (Major releases only)
+
+If all the steps above have been done you are good to go and create a new github release.
```

Comment:
Maybe we should recommend pushing a signed annotated tag, and then promote that to a GitHub release using the GitHub UI? WDYT?

Reply:
sound good, done